src/js/pouch.js

```diff
@@ -25,42 +46,79 @@ gpii.pouch.init = function (that) {
     fs.writeFileSync(that.options.pouchConfigPath, JSON.stringify(that.options.pouchConfig, null, 2));
 
     var MemPouchDB = PouchDB.defaults({ db: memdown });
-
-    fluid.each(that.options.databases, function (dbConfig, key) {
-        var db = new MemPouchDB(key);
-        if (dbConfig.data) {
-            var data = require(dbConfig.data);
-            db.bulkDocs(data);
-        }
-    });
-
     that.expressPouchdb = expressPouchdb(MemPouchDB, { configPath: pouchConfigPath });
 
-    //that.expressPouchdb.couchLogger.setFile(logPath);
+    if (PouchDB.isBeingCleaned) {
+        fluid.fail("I should never be allowed to load data if pouch is still being cleaned from the previous run...");
+    }
+    else {
+        var promises = [];
+        fluid.each(that.options.databases, function (dbConfig, key) {
+            var db = new MemPouchDB(key);
+            that.databaseInstances[key] = db;
+            if (dbConfig.data) {
+                var data = require(dbConfig.data);
+                promises.push(db.bulkDocs(data));
+            }
+        });
 
-    that.events.onStarted.fire();
+        when.all(promises).then(function () {
+            that.events.onStarted.fire();
+        });
+    }
 };
 
 gpii.pouch.getRouter = function (that) {
     return that.expressPouchdb;
 };
 
+// Remove all data from each database between runs, otherwise we encounter problems with data leaking between tests.
+//
+// https://github.com/pouchdb/pouchdb/issues/4124
+//
+gpii.pouch.cleanup = function (that) {
+    fluid.each(that.databaseInstances, function (db, key) {
+            db
+                .allDocs()
+                .then(function (result) {
+                    var bulkPayloadDocs = fluid.transform(result.rows, gpii.pouch.transformRecord);
+                    var bulkPayload     = {docs: bulkPayloadDocs};
+                    return db.bulkDocs(bulkPayload);
+                })
+                .then(function () {
+                    fluid.log("Deleted existing data from database '" + key + "'...");
+                    return db.compact();
+                })
+                .then(function (result) {
+                    fluid.log("Compacted existing database '" + key + "'...");
+                })
+                .catch(fluid.fail); // jshint ignore:line
 
-// TODO:  Write a change listener to allow easy adding of new databases
+        }
+    );
+};
 
-/*
-    The "databases" option is expected to be an array keyed by dbName, with options to control whether data is loaded or not, as in:
+gpii.pouch.transformRecord = function (record) {
+    // We cannot use "that" or its options here because we have already been destroyed by the time this function is called.
```

Comment:
This is problematic - since the expectation is that once a component has been destroyed, that all associated resources have been cleaned up. I'm not quite sure how the current test fixtures will be working, since it seems likely that one construction of pouch will interfere with the next one. I will try doing some testing to investigate. Given Infusion's model for component destruction is currently synchronous, I'm not sure there is a very clean way out. We'll have to talk a bit more.